.github/workflows/create_llvm.yml

```diff
@@ -45,6 +45,8 @@ jobs:
           - arch_flags: -DCMAKE_TOOLCHAIN_FILE="$GITHUB_WORKSPACE/ock/platform/arm-linux/aarch64-toolchain.cmake"
                         -DLLVM_HOST_TRIPLE=aarch64-unknown-linux-gnu
             arch: aarch64
+          - arch_flags: -DLLVM_BUILD_32_BITS=ON -DLIBXML2_LIBRARIES=IGNORE -DLLVM_ENABLE_TERMINFO=OFF
```

Comment:
This matches what we do on our other CI, but the second and third options are workaround for `LLVM_BUILD_32_BITS` being implemented in a way that doesn't handle everything that's needed, and these wouldn't be needed if we cross-compiled for 32-bit x86 in the same way that we do for AArch64. Maybe this is a good time to switch to that?

Does `LLVM_HOST_TRIPLE` need to be set too, or is that done automatically by `LLVM_BUILD_32_BITS`?

Reply:
`LLVM_HOST_TRIPLE` is set for cross builds because the value that would be used implicitly is wrong, but that applies to 32-bit vs. 64-bit builds as well. I checked our internal CI now, in our current build with `-DLLVM_BUILD_32_BITS=ON`, the log includes:
```
-- LLVM host triple: x86_64-unknown-linux-gnu
...
-- LLVM default target triple: x86_64-unknown-linux-gnu
```
We probably get away with it because we only use this build in limited ways, and those limited ways do always set the target triple, but it's still wrong, this is not the right host or target triple for 32-bit x86.

Reply:
if it's easy to do it in a cross way then sure, that makes sense - not sure how much Alan understands about it though.

Reply:
I assume the triple required is `x86-unknown-linux-gnu` - in which case I can add in the same way as is done for aarch64.

Reply:
It would be `i386-unknown-linux-gnu` or `i686-unknown-linux-gnu` depending on whether the build environment defaults to assuming newer processors. I do not know what the build environment in GitHub is like, @coldav do you know?

Reply:
I don't know.

Reply:
Okay, I checked, my local system's `gcc -v` includes `--with-arch-32=i686`, that's the Ubuntu-supplied GCC, and I think we use the Ubuntu-supplied GCC in GitHub too? It should be `i686-unknown-linux-gnu`, then.

Reply:
Confirmed for Github - thanks @hvdijk